grpc-ballerina/auth_client_self_signed_jwt_auth_handler.bal

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) 2021 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/jwt;
+
+# Represents JWT issuer configurations for JWT authentication.
+public type JwtIssuerConfig record {|
+    *jwt:IssuerConfig;
+|};
+
+# Defines the self signed JWT handler for client authentication.
+public class ClientSelfSignedJwtAuthHandler {
+
+    jwt:ClientSelfSignedJwtAuthProvider provider;
+
+    # Initializes the `grpc:ClientSelfSignedJwtAuthProvider` object.
+    #
+    # + config - The `grpc:JwtIssuerConfig` instance
+    public isolated function init(JwtIssuerConfig config) {
+        self.provider = new(config);
+    }
+
+    # Enrich the request with the relevant authentication requirements.
+    #
+    # + return - The JWT as a `string` or else an `grpc:ClientAuthError` in case of an error
+    public isolated function enrich() returns string|ClientAuthError {
```

Comment:
Shall we take the header map as an input argument?  and add the authentication header to it inside this function instead of returning the value

Reply:
Fixed in https://github.com/ballerina-platform/module-ballerina-grpc/pull/74/commits/b093610d4b430c74737c69bbe11b0e5d66abcfde